src/bench.c

```diff
@@ -95,7 +120,52 @@ int main(int argc, char** argv) {
     bench_verify_data data;
 
     int d = argc == 1;
-    int iters = get_iters(20000);
+    int default_iters = 20000;
+    int iters = get_iters(default_iters);
+
+    /* Check for invalid user arguments */
+    char* valid_args[] = {"ecdsa", "verify", "ecdsa_verify", "sign", "ecdsa_sign", "ecdh", "recover",
+                         "ecdsa_recover", "schnorrsig", "schnorrsig_verify", "schnorrsig_sign"};
+    size_t valid_args_size = sizeof(valid_args)/sizeof(valid_args[0]);
+    int invalid_args = have_invalid_args(argc, argv, valid_args, valid_args_size);
+
+    if (argc > 1) {
+        if (have_flag(argc, argv, "-h")
+           || have_flag(argc, argv, "--help")
+           || have_flag(argc, argv, "help")) {
+            help(default_iters);
+            return 0;
+        } else if (invalid_args) {
+            fprintf(stderr, "./bench: unrecognized argument.\n\n");
+            help(default_iters);
+            return 1;
+        }
+    }
+
+    /* Check if the user tries to benchmark optional module without building it */
+    #ifndef ENABLE_MODULE_ECDH
```

Comment:
I find it somewhat ugly that there is this much module-specific information inside `main()`, but I don't have a concrete suggestion on how to cleanly do it otherwise.

I'll try to address this in a follow-up PR if no simple solution can be used here.

Reply:
we can create a separate functions for these checks (like `int check_ecdh_enabled(argc,argv)`) in the `bench.h`. Not sure if this is a good approach.